### PR TITLE
added cmake for L476 nucleo and started board specific project

### PR DESCRIPTION
**Pull request by adrwstn**
added CMake for L476, under comp1-software/Example-L476/

### CODE REVIEW

**Comment by adrwstn** [comment]
I've added all HAL Drivers to CMake for L456, I figure this is okay (and possibly helpful) for a debug build config and we can remove the redundant ones if we want to set up a leaner build

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,9 @@ Makefile
 .mxproject
 startup_stm32f303xe.s
 STM32F303RETx_FLASH.ld
+
+# Added ignores for L476 - Dec 3rd
+startup_stm32l476xx.s
+arm-none-eabi-gcc.cmake
+STM32L476RGTx_FLASH.ld 
+cmake-build-debug/
```
Comment (Burke-Daniel) [inline .gitignore:15]: This is what CLion calls the build directory, alright to gitignore it but eventually we will have a script to automate builds, and the folder will be called `build/`. Also add newline to end of file

File: Example-L476/blink476/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pwr.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pwr_ex.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_cortex.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_exti.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pcd.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pcd_ex.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_can.c
+        # Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_ll_usb.c
+        # ----END OF LIST-------------------------
```
Comment (Burke-Daniel) [inline Example-L476/blink476/CMakeLists.txt:35]: Good to remove this, and the two that you have commented out below

File: Example-L476/blink476/Core/Inc/stm32l4xx_hal_conf.h [hunk 3]
```diff
@@ -0,0 +1,483 @@
+/**
```
Comment (Burke-Daniel) [inline Example-L476/blink476/Core/Inc/stm32l4xx_hal_conf.h:1]: So I don't think we will want to track this in git, see the other Blink project gitignores that live in Src and Inc that ignore these generated files

File: Examples/Blink/CMakeLists.txt [hunk 4]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.15.3)
```
Comment (Burke-Daniel) [inline Examples/Blink/CMakeLists.txt:1]: Would remove this from this PR, doesn't pertain to the l476 example

**Review by Burke-Daniel — CHANGES_REQUESTED** [review]
Good job on this. One thing we might wanna do is to start populating cubeMX with the peripherals that are configured on the pins in Jasper's schematic, but I'm ok to leave that out for now. Also, I think it it more appropriate for this example to live under `Examples/` as it's own project, instead of in a separate examples folder at the top level.

**Comment by Burke-Daniel** [comment]
@adrwstn have another look at this and lets get this closed out!

**Comment by adrwstn** [comment]
this is dead (for now)

**Pull request closed without merging**